src/parser/ast/type_argument_term.rs

```diff
@@ -0,0 +1,42 @@
+use crate::ir::ir_component::IRComponent;
+use crate::parser::ast::traits::ast_term::ASTTerm;
+
+pub struct TypeArgumentTerm {
+    value: i64,
+}
+
+impl TypeArgumentTerm {
+    pub fn new_boxed(value: String) -> Box<TypeArgumentTerm> {
+        Box::new(TypeArgumentTerm::new(value))
+    }
+
+    pub fn new(value: String) -> TypeArgumentTerm {
+        let value: i64 = value.parse().expect("Invalid syntax");
```

Comment:
Could we use syntax errors handling from lalrpop just to improve durability?
For instance:
```
Num: i32 = {
    r"[0-9]+" =>? i32::from_str(<>)
        .map_err(|_| ParseError::User {
            error: "number is too big"
        })
};
```
[source](http://lalrpop.github.io/lalrpop/tutorial/007_fallible_actions.html)